server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotResponse.java

```diff
@@ -86,4 +88,56 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
         builder.endObject();
         return builder;
     }
+
+    public static RestoreSnapshotResponse fromXContent(XContentParser parser) throws IOException {
+        RestoreSnapshotResponse response = new RestoreSnapshotResponse();
+
+        parser.nextToken(); // move to '{'
+
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new IllegalArgumentException("unexpected token [" + parser.currentToken() + "], expected ['{']");
+        }
+
+        parser.nextToken(); // move to 'snapshot' || 'accepted'
+
+        if ("snapshot".equals(parser.currentName())) {
+            response.restoreInfo = RestoreInfo.fromXContent(parser);
+        } else if ("accepted".equals(parser.currentName())) {
+            parser.nextToken(); // move to 'accepted' field value
+
+            if (parser.booleanValue()) {
+                // ensure accepted is a boolean value
```

Comment:
I think it is weird to just drop this on the floor. But we really don't have anywhere to put it.....

Reply:
Yes, this is caused by the (already existing) impedance mismatch of `XContent` and object structure which honestly causes quite some headaches.